app/lib/methods/getRoles.ts

```diff
@@ -88,25 +107,33 @@ export function getRoles(): Promise<void> {
 					const allRolesRecords = await rolesCollections.query().fetch();
 
 					// filter roles
-					const filteredRolesToCreate = roles.filter(i1 => !allRolesRecords.find(i2 => i1._id === i2.id));
-					const filteredRolesToUpdate = allRolesRecords.filter(i1 => roles.find(i2 => i1.id === i2._id));
+					const filteredRolesToCreate = roles.filter(i1 => !allRolesRecords.find(i2 => encodeRoleIdForStorage(i1._id) === i2.id));
+					const filteredRolesToUpdate = allRolesRecords.filter(i1 => roles.find(i2 => i1.id === encodeRoleIdForStorage(i2._id)));
 
 					// Create
 					const rolesToCreate = filteredRolesToCreate.map(role =>
 						rolesCollections.prepareCreate(
 							protectedFunction((r: TRoleModel) => {
-								r._raw = sanitizedRaw({ id: role._id }, rolesCollections.schema);
-								Object.assign(r, role);
+								r._raw = sanitizedRaw({ id: encodeRoleIdForStorage(role._id) }, rolesCollections.schema);
```

Comment:
Have you tried `sha256` instead of this `encodeRoleIdForStorage`? You won't need to decode it back, you can just compared to the encoded.